### DISCUSSION TITLE
Issue using .to_datetime on laps data frame

### DISCUSSION THREAD

**Discussion by laurence9899** [theOehrly/Fast-F1#451] [Q&A] [answered]
Hi,

I'm trying to create a histogram plot of lap times from a race to see their distribution.

I'm trying to use the below code to do it but keep getting the error 'object has no attribute 'to_datetime'

Any ideas on what may be wrong or how I can create the histogram plot?

import fastf1
import openpyxl
import numpy as np
import time
import pandas as pd

import fastf1.plotting
import seaborn as sns
from matplotlib import pyplot as plt

#Set up cache for race data from fastf1
fastf1.Cache.enable_cache('/Users/laurencehearne/Library/Mobile Documents/com~apple~CloudDocs/College/FYP/FastF1/')
race = fastf1.get_session(2023, 1, 'R')
race.load()

laps=race.laps.pick_quicklaps().reset_index()
plot_data = laps.LapTime.to_datetime(dailyData['time'], format="%D:%H:%M:%S.%f") 
print(plot_data)

hour_list = [t.hour for t in data]
bins = [0,9,17,24]

plt.hist(hour_list,bins)
plt.show()

Thank you

**Comment by Casper-Guo** [comment, marked as answer]
A minimally functional implementation:

```
import fastf1 as f
import fastf1.plotting as p
import pandas as pd
import matplotlib.pyplot as plt

race = f.get_session(2023, 1, 'R')
race.load()
p.setup_mpl()

laptimes = race.laps.pick_quicklaps()["LapTime"]

fig, ax = plt.subplots()
plt.hist(laptimes)
plt.show()
```
![image](https://github.com/theOehrly/Fast-F1/assets/89810860/eba7305e-f369-4b99-bea8-a3e79f25b3ff)

**Reply by laurence9899** [reply]
@Casper-Guo Thank you, this is exactly what I was trying to do.

**Comment by theOehrly** [comment]
@laurence9899 A note here, the LapTime is a timedelta object which is a duration (e.g. 1 hour). Therefore, it cannot be converted to a datetime object which is a specific point in time (e.g. 1st October 2023). 

Maybe you were aiming for this ``plot_data = laps.LapTime.dt.to_pytimedelta()``? (https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.dt.to_pytimedelta.html)

In any case, the solution by @Casper-Guo is likely what you want to use.

**Reply by laurence9899** [reply]
@theOehrly Thank you very much, I was struggling to understand that but now it makes sense.
